sdk/identity/Azure.Identity/src/AzureIdentityEventSource.cs

```diff
@@ -73,6 +74,7 @@ public void GetTokenSucceeded(string method, TokenRequestContext context, DateTi
             }
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode", Justification = "Parameters to this method are primitive and are trimmer safe.")]
```

Comment:
Curious - why are only these three events affected?

Reply:
The reason is because the other events all end up using the primitive overloads of WriteEvent, which aren't marked as `RequiresUnreferencedCode`. For example, there is a `WriteEvent` overload that takes 3 strings. Calling that doesn't produce a warning. But once you jump to 4 strings, there is no overload directly for that, and the compiler binds to the `params object[]` overload, which is marked as `RequiresUnreferencedCode`. Since we know that these are strings, it is safe to suppress the warning.

Note that this has been fixed when you target `net8.0` with https://github.com/dotnet/runtime/pull/83751. But since we aren't targeting `net8.0` (and 6 and 7 are still in support), we need to suppress these warnings so the user doesn't see them.